src/templates/template-common/utils.py

```diff
@@ -58,7 +58,21 @@ def setup_config(config_path, backend, **kwargs):
     return config
 
 
-#::: } else { :::#
+#::: } else if ((it.argparser == 'hydra')) { :::#
+
+
+def setup_config(config):
+    # To add a new attr in config from within the script
+    optional_attributes = ["backend", "num_iters_per_epoch"]
```

Comment:
Where comes from this `num_iters_per_epoch` parameter ?

Reply:
Basically the `main.py` content runs under `@hydra.main()` which does not support directly adding a new parameter like `num_iters_per_epoch` as defined in this line `config.num_iters_per_epoch = len(dataloader_train)`. So when we add this it gives an error like [here](https://github.com/pytorch-ignite/code-generator/actions/runs/5963666681/job/16177314813#step:17:51) . If we want it to be added, we have to force add it using this line `OmegaConf.update` or use `compose(config_name="config", overrides=[])`

Reply:
Can we make a copy of omega conf `config` to overcome this problem with `@hydra.main()` ?

Reply:
That doesn't seem to work as well, even if I use `deepcopy` it seems to give the same error